cpp/src/arrow/io/file.cc

```diff
@@ -152,20 +152,30 @@ static inline int64_t lseek64_compat(int fd, int64_t pos, int whence) {
 #endif
 }
 
+#if defined(_MSC_VER)
+static inline Status ConvertToUtf16(const std::string& input, std::wstring* result) {
+  if (result == nullptr) { return Status::Invalid("Pointer to result is not valid"); }
+
+  if (input.empty()) {
+    *result = std::wstring();
+    return Status::OK();
+  }
+
+  std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> utf16_converter;
+  *result = utf16_converter.from_bytes(input);
```

Comment:
Okay, sure, I will create separate ticket for that. I have seen other places ( https://github.com/apache/arrow/blob/master/cpp/src/arrow/io/file.cc#L134 ) without exception handling. It seems all of them should be covered at once.